datacube/model/__init__.py

```diff
@@ -718,8 +718,11 @@ def grid_range(lower, upper, step):
         >>> list(GridSpec.grid_range(1.0, 4.0, 3.0))
         [0, 1]
         """
-        assert step > 0.0
-        return range(int(math.floor(lower / step)), int(math.ceil(upper / step)))
+        assert abs(step) > 0.0
```

Comment:
This assert fails:
```python
assert list(GridSpec.grid_range(1.0, 4.0, -3.0)) == [-2, -1]
````
Having something like this an the top of the function should do the trick:
```python
if step < 0.0:
    lower, upper, step = -upper, -lower, -step
```